datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparisonJob.scala

```diff
@@ -0,0 +1,181 @@
+  private val conf: Config = ConfigFactory.load
+  private val errorColumnName: String = conf.getString("dataset-comparison.errColumn")
+  private val tmpColumnName: String = conf.getString("dataset-comparison.tmpColumn")
+  private val comparisonUniqueId: String = conf.getString("dataset-comparison.comparisonUniqueId")
+  private val actualPrefix: String = conf.getString("dataset-comparison.actualPrefix")
+  private val expectedPrefix: String = conf.getString("dataset-comparison.expectedPrefix")
+
+  def main(args: Array[String]): Unit = {
+    val cmd = DatasetComparisonConfig.getCmdLineArguments(args)
+    implicit val sparkSession: SparkSession = SparkSession.builder()
+      .appName(s"Dataset comparison - '${cmd.newPath}' and '${cmd.refPath}'")
+      .getOrCreate()
+
+    execute(cmd)
+  }
+
+  /**
+    * Execute the comparison
+    *
+    * @param cmd Provided configuration for the comparison
+    * @param sparkSession Implicit spark session
+    */
+  def execute(cmd: DatasetComparisonConfig)(implicit sparkSession: SparkSession): Unit = {
+    val dfReader: DataFrameReader = DataFrameReaderFactory.getDFReaderFromCmdConfig(cmd)
+    val expectedDf: DataFrame = dfReader.load(cmd.refPath)
+    val actualDf: DataFrame = dfReader.load(cmd.newPath)
+    val expectedSchema: StructType = getSchemaWithoutMetadata(expectedDf.schema)
+    val actualSchema: StructType = getSchemaWithoutMetadata(actualDf.schema)
+
+    if (cmd.hasKeysDefined) {
+      checkForDuplicateRows(actualDf, cmd.getKeys, cmd.outPath)
+    }
+
+    if (expectedSchema != actualSchema) {
+      val diffSchema = actualSchema.diff(expectedSchema) ++ expectedSchema.diff(actualSchema)
+      throw SchemasDifferException(cmd.refPath, cmd.newPath, diffSchema)
+    }
+
+    val expectedExceptActual: DataFrame = expectedDf.except(actualDf)
+    val actualExceptExpected: DataFrame = actualDf.except(expectedDf)
+
+    if (expectedExceptActual.count + actualExceptExpected.count == 0) {
+      scribe.info("Expected and actual data sets are the same.")
+    } else {
+      cmd.keys match {
+        case Some(keys) =>
+          handleKeyBasedErrorsPresent(keys, cmd.outPath, expectedExceptActual, actualExceptExpected)
+        case None =>
+          expectedExceptActual.write.format("parquet").save(s"${cmd.outPath}/expected_minus_actual")
+          actualExceptExpected.write.format("parquet").save(s"${cmd.outPath}/actual_minus_expected")
+      }
+      throw DatasetsDifferException(cmd.refPath, cmd.newPath, cmd.outPath, expectedDf.count(), actualDf.count())
+    }
+  }
+
+  /**
+    * Renames all columns expect the keys and append prefix to them.
+    *
+    * @param dataSet Dataset that needs columns renamed
+    * @param keys Keys that are not to be renamed
+    * @param prefix Prefix that will be put in front of column names
+    * @return New DataFrame with renamed columns
+    */
+  private def renameColumns(dataSet: DataFrame, keys: Seq[String], prefix: String): DataFrame = {
```

Comment:
```suggestion
  private def prefixColumns(dataSet: DataFrame, keys: Seq[String], prefix: String): DataFrame = {
```
The prefix is required and is the point of this function, so it seems more appropriate than rename (which sounds like an arbitrary name can be assigned)

Reply:
Changed as requested. Resolving comment.